[BUG] Flaky browser-DOM test: test_loads_full_page_on_a_slow_connection

### Setup

CI, `ubuntu-latest` runner, `uv run pytest -n auto`.

### What Happened

`tests/test_job_sidebar_scroll_dom.py::TestSidebarScroll::test_loads_full_page_on_a_slow_connection` failed with `assert 5 == 25` on run 33146106873, then passed on a rerun of the same commit. `main` was green on the surrounding runs, and the test passes locally, so this is timing, not a regression.

The fixture delays every batch by 800ms while the test calls `scroll_job_sidebar(settle_timeout=0.6)`. That combination is deliberate: 600ms < 800ms, so the first wait has to miss and the confirmation round at the full `settle_timeout` is what buys the batch. But it leaves only `2 * 600 - 800 = 400ms` of configured slack, and the failure is asymmetric. The budget is wall-clock (`Date.now()`), while the batch is a page `setTimeout`; when the runner starves the page's event loop the timer slips late and the budget does not. `assert 5 == 25` means zero successful rounds, so even the very first 1200ms window missed its 800ms timer.

The sibling tests are not exposed the same way. `test_survives_a_slow_first_batch` has `2 * 2000 - 2800 = 1200ms` of configured slack and one slow batch; this test has 400ms across four slow growth batches, so it has both the thinnest margin and the most chances to lose it.

### Steps to Reproduce

Load-dependent, and it does not reproduce on a 24-core machine (12/12 passes under 48 competing `yes` processes). The margin is what is reproducible: the fixture and the scroll loop are deterministic apart from timer scheduling, and the numbers above come straight from `settle_timeout`, the fixture's `delays`, and the two `waitForGrowth` calls in `linkedin_mcp_server/core/utils.py`.

### Logs

```
FAILED tests/test_job_sidebar_scroll_dom.py::TestSidebarScroll::test_loads_full_page_on_a_slow_connection - assert 5 == 25
```

--- a/tests/test_job_sidebar_scroll_dom.py
+++ b/tests/test_job_sidebar_scroll_dom.py
@@ -196,12 +196,23 @@ async def dom_page():
 
 class TestSidebarScroll:
     async def test_loads_full_page_on_a_slow_connection(self, dom_page):
-        """A batch slower than the old fixed 0.5s sleep must not end the loop."""
-        await dom_page.set_content(sidebar(total=25, batch=5, delays=[800]))
+        """A batch slower than the old fixed 0.5s sleep must not end the loop.
 
-        await scroll_job_sidebar(dom_page, settle_timeout=0.6)
+        Every batch is slower than ``settle_timeout``, so the first wait of
+        each round has to miss and the confirmation round at the full timeout
+        is what buys the batch. The margin that leaves is what the numbers are
+        chosen for: the budget is wall-clock while the batch is a page timer,
+        so a runner that starves the page's event loop slips the timer without
+        slowing the budget. At 800ms against ``settle_timeout=0.6``, each
+        of four growth batches had 400ms of configured slack, which failed
+        once in CI. At 1200ms against 1.0, two growth batches still require
+        the confirmation round while doubling that slack.
+        """
+        await dom_page.set_content(sidebar(total=15, batch=5, delays=[1200]))
 
-        assert await cards(dom_page) == 25
+        await scroll_job_sidebar(dom_page, settle_timeout=1.0)
+
+        assert await cards(dom_page) == 15
 
     async def test_loads_full_page_on_a_fast_connection(self, dom_page):
         """Without scrolling the rail stays at its initial five cards."""

